taipy/core/config/data_node_config.py

```diff
@@ -1121,10 +1127,13 @@ def _configure_s3_object(
 
         if aws_region is not None:
             properties[cls._OPTIONAL_AWS_REGION_PROPERTY] = aws_region
-        if aws_s3_object_parameters is not None:
-            properties[cls._OPTIONAL_AWS_S3_OBJECT_PARAMETERS_PROPERTY] = aws_s3_object_parameters
-
-        return cls.__configure(id, DataNodeConfig._STORAGE_TYPE_VALUE_S3_OBJECT, scope, validity_period, **properties)
+        if aws_s3_client_parameters is not None:
+            properties[cls._OPTIONAL_AWS_S3_CLIENT_PARAMETERS_PROPERTY] = aws_s3_client_parameters
+        if aws_s3_get_object_parameters is not None:
+            properties[cls._OPTIONAL_AWS_S3_GET_OBJECT_PARAMETERS_PROPERTY] = aws_s3_get_object_parameters
+        if aws_s3_put_object_parameters is not None:
+            properties[cls._OPTIONAL_AWS_S3_PUT_OBJECT_PARAMETERS_PROPERTY] = aws_s3_put_object_parameters
+            return cls.__configure(id, DataNodeConfig._STORAGE_TYPE_VALUE_S3_OBJECT, scope, validity_period, **properties)
```

Comment:
```suggestion

        return cls.__configure(id, DataNodeConfig._STORAGE_TYPE_VALUE_S3_OBJECT, scope, validity_period, **properties)
```

There should not be a tab here